packages/zoe/src/contractFacet/zcfZygote.js

```diff
@@ -264,15 +264,17 @@ export const makeZCFZygote = (
     handleOffer: (invitationHandle, zoeSeatAdmin, seatData) => {
       const zcfSeat = makeZCFSeat(zoeSeatAdmin, seatData);
       const offerHandler = takeOfferHandler(invitationHandle);
-      const offerResultP = E(offerHandler)(zcfSeat).catch(reason => {
-        if (reason === undefined) {
-          const newErr = new Error(
-            `If an offerHandler throws, it must provide a reason of type Error, but the reason was undefined. Please fix the contract code to specify a reason for throwing.`,
-          );
-          throw zcfSeat.fail(newErr);
-        }
-        throw zcfSeat.fail(reason);
-      });
+      const offerResultP = E(offerHandler)(zcfSeat, seatData.offerArgs).catch(
+        reason => {
+          if (reason === undefined) {
+            const newErr = new Error(
+              `If an offerHandler throws, it must provide a reason of type Error, but the reason was undefined. Please fix the contract code to specify a reason for throwing.`,
```

Comment:
This phrasing suggests that we enforce that reason is of type Error, rather than merely that it is undefined.

Reply:
Hmm, I think we do want to enforce that the reason is of type Error. Making sure it's not undefined is part of that. Can we do better?

This code didn't change in this PR, so I think we keep this discussion going separately, after this PR gets merged.